src/lambda/vue/samInvoke.ts

```diff
@@ -450,3 +426,47 @@ function getLaunchConfigQuickPickItems(launchConfig: LaunchConfiguration, uri: v
             }
         })
 }
+
+function pruneConfig(config: AwsSamDebuggerConfiguration, name: string): AwsSamDebuggerConfiguration {
+    const newConfig = pruneConfigHelper(config)
+    newConfig.name = name
+
+    if (isTemplateTargetProperties(config.invokeTarget)) {
+        newConfig.invokeTarget = {
+            target: config.invokeTarget.target,
+            logicalId: config.invokeTarget.logicalId,
+            templatePath: config.invokeTarget.templatePath,
+        }
+    } else if (isCodeTargetProperties(config.invokeTarget)) {
+        newConfig.invokeTarget = {
+            target: config.invokeTarget.target,
+            lambdaHandler: config.invokeTarget.lambdaHandler,
+            projectRoot: config.invokeTarget.projectRoot,
+        }
+    }
+
+    return newConfig
+}
+
+function pruneConfigHelper(object: { [key: string]: any }): any | undefined {
```

Comment:
I would call this `pruneConfigRecursive` or `traverseAndPrune`. That avoids a meaningless "helper" name :)

Another common convention is to prefix "do" on the function that actually performs the work of the outer interface. In this case  `pruneConfig` is the interface, so `pruneConfigHelper` would be named `doPruneConfig`.